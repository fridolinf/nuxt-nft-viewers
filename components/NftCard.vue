<template>
  <div
    id="nft-card"
    v-for="(nft, i) in cardData as MoralisMetadata[]"
    :key="i"
    class="m-3 shadow-lg shadow-black rounded-md p-2 bg-white"
  >
    <div class="relative">
      <div
        v-show="nft && nft.image ? nft.image : '/assets/images/noimg.png'"
        class="bg-cyan-300 w-32 h-52 absolute md:-right-5 xl:left-2/3 -z-10 rounded-md"
      ></div>
    </div>
    <p class="text-center font-bold">
      {{ nft && nft.collectionName ? nft.collectionName : "" }}
    </p>
    <nuxt-img
      quality="50"
      class="rounded-xl shadow-sm shadow-cyan-500 my-2 w-full"
      loading="lazy"
      :src="
        nft && nft.image
          ? nft.image.replace('ipfs://', 'https://cloudflare-ipfs.com/ipfs/')
          : '../public/assets/images/noimg.png'
      "
      :alt="nft && nft.name ? nft.name : 'nft-image'"
    />
    <div class="text-clip">
      <p>Metadata</p>
      <p class="font-semibold">{{ nft.name }}</p>
      <p>
        {{
          nft && nft.description
            ? nft.description.substring(0, 150) + " " + "..."
            : ""
        }}
      </p>
    </div>
  </div>
</template>

<script setup lang="ts">
import { MoralisMetadata } from "../interfaces/IMoralis";

defineProps({
  cardData: { type: Array, required: true },
});
</script>
