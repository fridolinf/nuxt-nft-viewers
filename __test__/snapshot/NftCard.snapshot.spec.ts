import NftCard from "../../components/NftCard.vue";
import { mount } from "@vue/test-utils";

describe("Snapshot Index Page", () => {
  test("Snapshot index page component", () => {
    const pages = mount(NftCard);
    expect(pages.html()).toMatchSnapshot();
  });
});
