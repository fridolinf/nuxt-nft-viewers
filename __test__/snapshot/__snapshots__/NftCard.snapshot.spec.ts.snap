// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Index Page Snapshot index page component 1`] = `""`;
